lib/gui/etcher.js

```diff
@@ -40,14 +49,19 @@ const configUrl = settings.get('configUrl') || 'https://balena.io/etcher/static/
 const checkForUpdates = async (interval) => {
   // We use a while loop instead of a setInterval to preserve
   // async execution time between each function call
-  while (true) {
-    try {
-      const release = await autoUpdater.checkForUpdates()
-      if (release.updateInfo.stagingPercentage) {
-        await autoUpdater.downloadUpdate()
+  while (!packageUpdated) {
+    if (settings.get('updatesEnabled')) {
+      try {
+        const release = await autoUpdater.checkForUpdates()
+        const isOutdated = semver.compare(release.updateInfo.version, version) > 0
+        const shouldUpdate = parseInt(release.updateInfo.stagingPercentage, 10) > 0
+        if (shouldUpdate && isOutdated && !packageUpdated) {
```

Comment:
No need to check for `!packageUpdated` here, it's already `false` if we're in this `while` loop.